### PR TITLE
docs: consolidate deployment modes guides

### LINKED ISSUE #8715: docs: consolidate deployment modes guides
### Current State

The [deployment modes documentation page](https://apisix.apache.org/docs/apisix/deployment-modes/) contains documentation about all the deployment modes. There is also another page for [stand-alone mode](https://apisix.apache.org/docs/apisix/stand-alone/) with just its documentation.

### Desired State

Instead, these two pages should be merged to a single page and moved out of the "Others" section.

### PR DESCRIPTION
**Pull request by Neilblaze**
### Description

> The [deployment modes documentation page](https://apisix.apache.org/docs/apisix/deployment-modes/) contains documentation about all the deployment modes. There is also another page for [stand-alone mode](https://apisix.apache.org/docs/apisix/stand-alone/) with just its documentation. So, these two pages should be merged into a single page and moved out of the "Others" section.

Hence the entire content from the `stand-alone-mode` page has been scraped out & appended to the `deployment modes documentation page` in the pre-existing `stand-alone` section at the bottom of the same. The page is then shifted outside of the "Others" section. Moreover, the single page which was initially preserved for `stand-alone` mode is removed, as it's no longer required once the merging of both is done!

**P.S. >** There will be no change in the "zh" docs, since the file name remains the same & only update processed is for "en". For example, the "Standalone" remains as `../../en/latest/deployment-modes.md#standalone`

Fixes #8715

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
Also requesting @nfrankel for review.

**Comment by navendu-pottekkat** [comment]
@Neilblaze Could you check the lint error?

**Comment by Neilblaze** [comment]
@navendu-pottekkat should be fixed now!

**Comment by nfrankel** [comment]
@navendu-pottekkat  Do we have a preview of the generate site?

**Comment by navendu-pottekkat** [comment]
> @navendu-pottekkat Do we have a preview of the generate site?

@nfrankel No. Each of the APISIX projects contain its own documentation (.md files) and the apisix-website repo pulls these files and builds the site on release. So, only the apisix-website repo builds the actual site. It is hard for the reviewers unless you manually pull the changes to a local clone of the apisix-website and build the site yourself. The team discussed this issue a long time back and the I proposed to move all docs to one repo. I don't know if there are any other efforts to make this process easier. @SkyeYoung may have some info.

**Comment by nfrankel** [comment]
@navendu-pottekkat I'll check next week then, if it can wait until then

File: docs/en/latest/deployment-modes.md [hunk 1]
```diff
@@ -155,15 +155,273 @@ deployment:
...
+This method is more suitable for two types of users:
 
-To configure APISIX in standalone mode:
+1. kubernetes(k8s)：Declarative API that dynamically updates the routing rules with a full yaml configuration.
+2. Different configuration centers: There are many implementations of the configuration center, such as Consul, etc., using the full yaml file for intermediate conversion.
 
-```yaml title="conf/config.yaml"
+The routing rules in the `conf/apisix.yaml` file are loaded into memory immediately after the APISIX node service starts. And every time interval (default 1 second), will try to detect whether the file content is updated, if there is an update, reload the rule.
+
+*Note*: When reloading and updating routing rules, they are all hot memory updates, and there will be no replacement of working processes, it is a hot update.
```
Comment (nfrankel) [inline docs/en/latest/deployment-modes.md:167]: _Note_: Reloading and updating routing rules are all hot memory updates. There is no replacement of working processes, it's a hot update.

File: docs/en/latest/deployment-modes.md [hunk 2]
```diff
@@ -155,15 +155,273 @@ deployment:
...
+
+Since the current Admin API is based on the etcd configuration center solution, enable Admin API is not allowed when the Stand-alone mode is enabled.
+
+The Stand-alone mode can only be enabled when set the role of APISIX as data plane. We can set `deployment.role` to `data_plane` and `deployment.role_data_plane.config_provider` to `yaml`.
```
Comment (nfrankel) [inline docs/en/latest/deployment-modes.md:171]: Standalone mode can only be enabled when we set the role of APISIX as data plane. We set `deployment.role` to `data_plane` **and** `deployment.role_data_plane.config_provider` to `yaml`.

File: docs/en/latest/deployment-modes.md [hunk 3]
```diff
@@ -155,15 +155,273 @@ deployment:
...
-    role_data_plane:
-       config_provider: yaml
+  role: data_plane
+  role_data_plane:
+    config_provider: yaml
+```
+
+### How to configure rules
+
+All of the rules are stored in one file which named `conf/apisix.yaml`,
```
Comment (nfrankel) [inline docs/en/latest/deployment-modes.md:184]: All of the rules are stored in one file named `conf/apisix.yaml`;
APISIX checks if this file has any change **every second**.
If the file changed and ends with `#END`,
APISIX loads the rules from this file and updates its memory.

File: docs/en/latest/deployment-modes.md [hunk 4]
```diff
@@ -155,15 +155,273 @@ deployment:
...
+  -
+    uri: /hello
+    upstream:
+        nodes:
+            "127.0.0.1:1980": 1
+        type: roundrobin
+#END
+```
+
+*NOTE*: APISIX will not load the rules into memory from file `conf/apisix.yaml` if there is no `#END` at the end.
```
Comment (nfrankel) [inline docs/en/latest/deployment-modes.md:202]: *WARNING*: APISIX will not load the rules into memory from file `conf/apisix.yaml` if there is no `#END` at the end.

**Review by nfrankel — CHANGES_REQUESTED** [review]
* OK with the general structure
* Please use Grammarly, I found a couple of not-so-proper English
* Technical requirements that aren't user-friendly should be highlighted:

    1. `#END` at the end
    2. The `role` must be `data_plane`

**Comment by Neilblaze** [comment]
@nfrankel Thanks for the update. I'm on it!

**Comment by Neilblaze** [comment]
Lemme know if any modification is required on the same :)
cc : @nfrankel & @navendu-pottekkat

File: docs/en/latest/deployment-modes.md [hunk 5]
```diff
@@ -70,20 +71,20 @@ The instance of APISIX deployed as the traditional role will:
 
 ## Decoupled
 
-In the decoupled deployment mode the data plane and control plane instances of APISIX are deployed separately. i.e one instance of APISIX is configured to be a data plane and the other to be a control plane.
+In the decoupled deployment mode the `data_plane` and `control_plane` instances of APISIX are deployed separately, i.e one instance of APISIX is configured to be a *data plane* and the other to be a *control plane*.
```
Comment (nfrankel) [inline docs/en/latest/deployment-modes.md:74]: In the decoupled deployment mode the `data_plane` and `control_plane` instances of APISIX are deployed separately, _i.e._, one instance of APISIX is configured to be a *data plane* and the other to be a *control plane*.

Comment (Neilblaze) [reply docs/en/latest/deployment-modes.md:74]: @nfrankel I'm sorry, but can you please highlight the required changes? I'm not able to see it. 😅
Also, I did use Grammarly to rectify most of the grammatical mistakes, so I think it should be fine now.😄

Comment (nfrankel) [reply docs/en/latest/deployment-modes.md:74]: There should be a command before and after "i.e." (and "e.g."). Grammarly is adamant about it.

Comment (Neilblaze) [reply docs/en/latest/deployment-modes.md]: Ah, I see now. Just rectified it, can you have a look at whether it's ready to merge?

**Review by nfrankel — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Review by navendu-pottekkat — APPROVED** [review]

**Pull request merged**